aten/src/ATen/TensorImpl.h

```diff
@@ -23,6 +25,7 @@ struct TensorImpl {
   virtual Scalar localScalar() = 0;
   virtual void assign_(Scalar s) = 0;
   virtual void * unsafeGetTH(bool retain) = 0;
+  virtual std::unique_ptr<Storage> storage() = 0;
```

Comment:
Is `Storage` a smart wrapper like `Tensor`? If yes, why return a `unique_ptr` and not the wrapper?

Reply:
No, it's a more typical C++ class

Reply:
Are you serious...

Reply:
Still, making it a unique_ptr is a bit weird, because it usually means that this is the thing that owns the object and all the state it's managing, while it's not the case here. But I guess that's just how it is in ATen, so feel free to merge if you want.